crash at cmd/swarmctl/common/resolver.go

https://github.com/docker/swarm-v2/blob/master/cmd/swarmctl/common/resolver.go#L37 doesn't handle nil pointer `return res.Node.Description.Hostname`. 

```
azureuser@swarm-master-2 ~/swarmv2 $ sudo ./swarmctl task inspect 3jmnouju5wdw6wvl5gt0yv8sv
ID                     : 3jmnouju5wdw6wvl5gt0yv8sv
Instance               : 0
Service                : cadvisor
Status
  Desired State        : RUNNING
  Last State           : PREPARING
  Timestamp            : 2016-06-03T00:58:08.821798906Z
  Message              : preparing
Node                   : swarm-node-3
Spec
  Image                : google/cadvisor:v0.22.0

===> Instance History
Task ID                      Service     Instance    Image                      Desired State    Last State                 Node
-------                      -------     --------    -----                      -------------    ----------                 ----
3jmnouju5wdw6wvl5gt0yv8sv    cadvisor    0           google/cadvisor:v0.22.0    RUNNING          PREPARING 8 minutes ago    swarm-node-3
4sd6g1ffl55felv7wzhecaq40    cadvisor    0           google/cadvisor:v0.22.0    SHUTDOWN         REJECTED 8 minutes ago     swarm-node-3
8e9k7vh679dyz3qwhnvlbx9ns    cadvisor    0           google/cadvisor:v0.22.0    SHUTDOWN         REJECTED 8 minutes ago     swarm-node-3
e4bo0xmqdt8hofnd3fkwopbma    cadvisor    0           google/cadvisor:v0.22.0    SHUTDOWN         REJECTED 8 minutes ago     swarm-node-3
b36vhi2omy39h8r12z84lsr06    cadvisor    0           google/cadvisor:v0.22.0    SHUTDOWN         REJECTED 8 minutes ago     swarm-node-3
6gllfks4tbnsnxiwb16rz5fxk    cadvisor    0           google/cadvisor:v0.22.0    SHUTDOWN         REJECTED 8 minutes ago     swarm-node-3
8o1t6yg6oe4dt2dqbl73xodbv    cadvisor    0           google/cadvisor:v0.22.0    SHUTDOWN         REJECTED 8 minutes ago     swarm-node-3
161tq1a565f8ishyqldmy45vk    cadvisor    0           google/cadvisor:v0.22.0    RUNNING          ALLOCATED 8 minutes ago    swarm-node-3
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x63a73d]

goroutine 1 [running]:
panic(0xc525e0, 0xc8200120d0)
    /usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/docker/swarm-v2/cmd/swarmctl/common.(*Resolver).get(0xc820105be0, 0xcc2e40, 0xc820105868, 0xc8201adc60, 0xd, 0x0, 0x0)
    /home/dchen/go/src/github.com/docker/swarm-v2/cmd/swarmctl/common/resolver.go:37 +0x1cd
github.com/docker/swarm-v2/cmd/swarmctl/common.(*Resolver).Resolve(0xc820105be0, 0xcc2e40, 0xc820105868, 0xc8201adc60, 0xd, 0x0, 0x0)
    /home/dchen/go/src/github.com/docker/swarm-v2/cmd/swarmctl/common/resolver.go:59 +0x159
github.com/docker/swarm-v2/cmd/swarmctl/task.Print(0xc8201b2700, 0x9, 0x10, 0x1, 0xc820105be0)
    /home/dchen/go/src/github.com/docker/swarm-v2/cmd/swarmctl/task/print.go:60 +0x439
github.com/docker/swarm-v2/cmd/swarmctl/task.glob.func1(0x11ab860, 0xc820169d70, 0x1, 0x1, 0x0, 0x0)
    /home/dchen/go/src/github.com/docker/swarm-v2/cmd/swarmctl/task/inspect.go:99 +0x78e
github.com/docker/swarm-v2/vendor/github.com/spf13/cobra.(*Command).execute(0x11ab860, 0xc820169d00, 0x1, 0x1, 0x0, 0x0)
    /home/dchen/go/src/github.com/docker/swarm-v2/vendor/github.com/spf13/cobra/command.go:565 +0x62c
github.com/docker/swarm-v2/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x11aca60, 0x11ab860, 0x0, 0x0)
    /home/dchen/go/src/github.com/docker/swarm-v2/vendor/github.com/spf13/cobra/command.go:656 +0x55c
main.main()
    /home/dchen/go/src/github.com/docker/swarm-v2/cmd/swarmctl/main.go:21 +0x37
```

--- a/cmd/swarmctl/common/resolver.go
+++ b/cmd/swarmctl/common/resolver.go
@@ -34,6 +34,9 @@ func (r *Resolver) get(t interface{}, id string) string {
 		if res.Node.Spec.Annotations.Name != "" {
 			return res.Node.Spec.Annotations.Name
 		}
+		if res.Node.Description == nil {
+			return id
+		}
 		return res.Node.Description.Hostname
 	case api.Service:
 		res, err := r.c.GetService(r.ctx, &api.GetServiceRequest{ServiceID: id})

--- a/cmd/swarmctl/node/inspect.go
+++ b/cmd/swarmctl/node/inspect.go
@@ -27,8 +27,9 @@ func printNodeSummary(node *api.Node) {
 	}
 	common.FprintfIfNotEmpty(w, "ID\t: %s\n", node.ID)
 	common.FprintfIfNotEmpty(w, "Name\t: %s\n", spec.Annotations.Name)
-	common.FprintfIfNotEmpty(w, "Hostname\t: %s\n", node.Description.Hostname)
-
+	if node.Description != nil {
+		common.FprintfIfNotEmpty(w, "Hostname\t: %s\n", node.Description.Hostname)
+	}
 	fmt.Fprintln(w, "Status:\t")
 	common.FprintfIfNotEmpty(w, "  State\t: %s\n", node.Status.State.String())
 	common.FprintfIfNotEmpty(w, "  Message\t: %s\n", node.Status.Message)

